exercise-book/src/nrf52-radio-setup.md

```diff
@@ -12,4 +12,4 @@ received 5 bytes (CRC=Ok(0xdad9), LQI=53)
 
 The program broadcasts a radio packet that contains the 5-byte string `Hello` over channel 20 (which has a center frequency of 2450 MHz). The `loopback` program running on the Dongle is listening to all packets sent over channel 20; every time it receives a new packet it reports its length and the Link Quality Indicator (LQI) metric of the transmission over the USB/serial interface. As the name implies the LQI metric indicates how good the connection between the sender and the receiver is.
 
-To repeatedly send a packet without re-flashing your DK, try pressing the BOOT/RESET button (next to the nRF USB connector that we aren't using yet).
+Because of how our firmware generates a *semihosting exception* to tell our flashing tool (`probe-run`) when the firmware has finished running, if you load the `radio-send` firmware and then power-cycle the nRF52840-DK, the firmware will enter a reboot loop and repeatedly send a packet. This is because nothing catches the *semihosting exception* and so the CPU reboots, sends a packet, and then tries another *semihosting exception*.
```

Comment:
```suggestion
Because of how our firmware generates a *semihosting exception* to tell our flashing tool (`probe-run`) when the firmware has finished running, if you load the `radio-send` firmware and then power-cycle the nRF52840-DK, the firmware will enter a reboot loop and repeatedly send a packet. This is because nothing catches the *semihosting exception* and so the CPU reboots, sends a packet, and then tries another *semihosting exception*.

Make sure the firmware you flashed corresponds to the name shown after `cargo xtask usb-list` - in this case, it should say `puzzle-fw`. Should it not, run `cargo run` in the `nrf52-code/puzzle-fw` crate when the Dongle is in bootloader bmode to flash the `puzzle-fw` firmware onto it.
```

Reply:
Unfortunately compiling the puzzle-fw yourself won't get you the secret message, because it's stored in a Github Secret. You have to use the pre-compiled binary.

Reply:
But I opened #88 to track checking which firmware is installed